mesonbuild/interpreter.py

```diff
@@ -2782,6 +2782,9 @@ def find_program_impl(self, args, native=False, required=True, silent=True):
             progobj = self.program_from_file_for(for_machine, args, silent=silent)
         if progobj is None:
             progobj = self.program_from_system(args, silent=silent)
+        if progobj is None and args[0].endswith('python3'):
+            prog = dependencies.ExternalProgram('python3', mesonlib.python_command, silent=True)
+            progobj = ExternalProgramHolder(prog)
```

Comment:
Hmm, the problem is not `find_program('python3')` but `find_program('some-script.py')` where `some-script.py` is `#!/usr/bin/env python3`.